packages/flutter/lib/src/widgets/implicit_animations.dart

```diff
@@ -362,10 +364,9 @@ abstract class ImplicitlyAnimatedWidgetState<T extends ImplicitlyAnimatedWidget>
   @override
   void didUpdateWidget(T oldWidget) {
     super.didUpdateWidget(oldWidget);
-    if (widget.curve != oldWidget.curve)
-      _updateCurve();
+    final bool curveChanged = widget.curve != oldWidget.curve;
     _controller.duration = widget.duration;
-    if (_constructTweens()) {
+    if (_constructTweens() || (_controller.isAnimating && curveChanged)) {
```

Comment:
changing the curve shouldn't require the tween to update, what am i missing?

Reply:
When you change the curve while the animation is running it previously jumped to the new value on the new curve. With this change, changing the curve will be smooth: It will continue the animation at the current animation value and smoothly animate to the end using the new curve.For that, we need to update the tween to have the current animation value as the new begin.

Reply:
I don't feel strongly about this change, but it seems to be more in the spirit of the widget, which always promises smooth animations with no jumps.

Reply:
As discussed, I am removing this change from this PR and create a separate PR with a breaking change where the new curve gets applied the next time the targetValue changes.